src/server/search/search_family.cc

```diff
@@ -343,11 +346,32 @@ void SearchFamily::FtCreate(CmdArgList args, ConnectionContext* cntx) {
   if (auto err = parser.Error(); err)
     return (*cntx)->SendError(err->MakeReply());
 
+  cntx->transaction->Schedule();
+
+  // Check if index already exists
+  atomic_uint exists_cnt = 0;
+  cntx->transaction->Execute(
+      [idx_name, &exists_cnt](auto* tx, auto* es) {
+        if (es->search_indices()->GetIndex(idx_name) != nullptr)
+          exists_cnt.fetch_add(1, std::memory_order_relaxed);
+        return OpStatus::OK;
+      },
+      false);
+
+  DCHECK(exists_cnt == 0u || exists_cnt == shard_set->size());
```

Comment:
Not that it matters greatly, but since it's supposed to either be `0` of `shard_set->size()`, you could do it in a single hop, no?